src/charm.py

```diff
@@ -344,6 +382,9 @@ def _update(self, event):
                 "VISIBILITY_PORT": visibility_conn["port"],
                 "VISIBILITY_USER": visibility_conn["user"],
                 "VISIBILITY_PSWD": visibility_conn["password"],
+                "TEMPORAL_BROADCAST_ADDRESS": str(self.model.get_binding("peer").network.bind_address),
+                # TODO(kelkawi-a): do not assume the app is always deployed with this name.
```

Comment:
Isn't getting the name dynamically simple enough?

Reply:
So as it stands currently, the two services (matching and worker) must talk to the frontend service (which is deployed using the name ```temporal-k8s```). With a scaled deployment, the four services are deployed as four different applications (which are related and communicate at the database level, not at the juju level). Therefore, we would need to figure out a way of getting the ```temporal-k8s``` charm to communicate its name with the other two charms deployed for them to correctly communicate with it. 

TLDR: ```PUBLIC_FRONTEND_ADDRESS``` config is being set to ```temporal-k8s:7233``` to allow the matching and worker services to communicate with it.